src/kOS/Module/kOSProcessor.cs

```diff
@@ -528,8 +528,15 @@ public void SetMode(ProcessorModes newProcessorMode)
                 switch (newProcessorMode)
                 {
                     case ProcessorModes.READY:
-                        if ((ProcessorMode == ProcessorModes.STARVED || ProcessorMode == ProcessorModes.OFF) && shared.Cpu != null)
-                            shared.Cpu.Boot();
+                        if (Config.Instance.StartOnArchive)
```

Comment:
@erendrake looks like I may have missed an edit to your comment, or I failed to read it all the way through.  I still think that `if (ProcessorMode != ProcessorModes.READY)` is redundant.

The question I have is how do we want to future proof the code?  If it is more likely that a mode is added which does not require running the boot method (for instance a sleep mode) then the code should check against known transitions that do require it (thus reverting the line I edited).  If it is more likely that a mode is added which will still require the boot method to be called (like some kind of docked/undocked mode), then I think it would be best to leave it as it is in my commit.

I'm not sure if that makes any sense. I can re-word it tomorrow if needed.  I'm kind of tired right now.

I guess that means my vote is either
`if ((ProcessorMode == ProcessorModes.STARVED || ProcessorMode == ProcessorModes.OFF) && shared.Cpu != null)`
or
`if (shared.Cpu != null)`
Let me know which way you want to go with it, and I'll update the commit to whatever your choice is.  The part I thought was important was to switch back to the default volume (it's pretty important with boot scripts, since booting with the archive selected will launch boot.ks if it is stored in the script folder).